### ISSUE TITLE
Interpreting predictions in the variational autoencoder example

### ISSUE THREAD

**Issue by mjg4243** [rstudio/keras#522]
From here:  https://github.com/rstudio/keras/blob/master/vignettes/examples/variational_autoencoder.R

'Visualization' shows predictions using the encoder as ordered pairs, matched in sequence to labels for test labels, then plotting.  Result is a nice kind of cluster display -- but how does the predictor output relate to the power of the classifier?

**Comment by mjg4243** [comment]
I appreciate your taking the trouble to reply.  
I understand dense models pretty well, but am trying to find out if I can do better with a tough classification problem using the autoencoder.
Having thought about it some more, and found some interesting information --
Stacked Autoencoders - Ufldl


  
|  
|   
|   
|   |    |

   |

  |
|  
|   |  
Stacked Autoencoders - Ufldl
   |   |

  |

  |

 
It looks as if I should be able to get at the latent space learning with a straightforward model.  In the Keras information it looks like the 'encoder' model, which I listed out, provides the extra information the autoencoder provides (??).  I would think I could, somehow, just append another layer breaking out applicable probabilities associated with each of the classes --  that is, an assignment of class to each of the test data items.  Now it is still a bit mysterious, to me, at least, that it can be made to work but that is what I get from the stacked autoencoders information.
Any further coments?  I would much appreciate them.
Best regards Mike Gallagher

**Comment by skeydan** [comment]
Hi,

seems like something went wrong with the rendering on github... I cannot see the copy-pasted part you're referring to (stacked autoencoders?)

**Comment by mjg4243** [comment]
Maybe this will come over in better shape. Mike Gallagher

**Comment by mjg4243** [comment]
[stacked.pdf](https://github.com/rstudio/keras/files/2366576/stacked.pdf)

**Comment by skeydan** [comment]
Yes, right, the `encoder` is a mapping from input data to latent space.

You could extract and save its output, and use that as input to a classifier, just like you can extract image features from models like VGG or Inception ...

Here

https://blogs.rstudio.com/tensorflow/posts/2018-01-24-keras-fraud-autoencoder/

is an interesting example using autoencoders (in general, not necessarily VAE) for anomaly detection.

**Comment by mjg4243** [comment]
Thanks!  I will go through it. Mike Gallagher

**Comment by skeydan** [comment]
OK, great!
